components/places/src/storage/bookmarks.rs

```diff
@@ -1236,6 +1236,170 @@ pub fn fetch_tree(db: &impl ConnExt, item_guid: &SyncGuid) -> Result<Option<Book
     Ok(Some(root))
 }
 
+/// This is similar to fetch_tree, but does not recursively fetch children of
+/// folders.
+///
+/// If `get_direct_children` is true, it will return 1 level of folder children,
+/// otherwise it returns just their guids.
+///
+/// It also produces the protobuf message type directly, rather than
+/// add a special variant of this bookmark type just for this function.
+pub fn fetch_bookmark(
+    db: &impl ConnExt,
+    item_guid: &SyncGuid,
+    get_direct_children: bool,
+) -> Result<Option<ProtoBookmark>> {
+    let _tx = db.unchecked_transaction()?;
+    let bookmark = fetch_bookmark_in_tx(db, item_guid, get_direct_children)?;
+    // Note: We let _tx drop (which means it does a rollback) since it doesn't
+    // matter, we just are using a transaction to ensure things don't change out
+    // from under us, since this executes more than one query.
+    Ok(bookmark)
+}
+
+// Implementation of fetch_bookmark
+fn fetch_bookmark_in_tx(
```

Comment:
Yeah, it wouldn't be too hard to add, I just wanted to avoid more complexity than needed in this patch. I'm going to lie to myself and imagine that the performance hit doesn't matter that much because we're in a transaction.

Reply:
Agree it's not worth doing now, but maybe add lina's note as a comment?

Reply:
I did. I've also added a list of things I'm going to file a follow up for at in the PR description, which includes this.